providers/asrockrack/asrockrack.go

```diff
@@ -0,0 +1,304 @@
+package asrockrack
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib/internal/httpclient"
+	"github.com/bmc-toolbox/bmclib/providers"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the provider implementation
+	ProviderName = "asrockrack"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "vendorapi"
+)
+
+var (
+	// Features implemented by asrockrack https
+	Features = registrar.Features{
+		providers.FeatureBiosVersionRead,
+		providers.FeatureBmcVersionRead,
+		providers.FeatureBiosFirmwareUpdate,
+		providers.FeatureBmcFirmwareUpdate,
+	}
+)
+
+// ASRockRack holds the status and properties of a connection to a asrockrack bmc
+type ASRockRack struct {
+	ip            string
+	username      string
+	password      string
+	loginSession  *loginSession
+	httpClient    *http.Client
+	fwInfo        *firmwareInfo
+	resetRequired bool // Indicates if the BMC requires a reset
+	skipLogout    bool // A Close() / httpsLogout() request is ignored if the BMC was just flashed - since the sessions are terminated either way
+	log           logr.Logger
+}
+
+// New returns a new ASRockRack instance ready to be used
+func New(ctx context.Context, ip string, username string, password string, log logr.Logger) (*ASRockRack, error) {
+
+	client, err := httpclient.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ASRockRack{
+		ip:           ip,
+		username:     username,
+		password:     password,
+		log:          log,
+		loginSession: &loginSession{},
+		httpClient:   client,
+	}, nil
+}
+
+// Compatible implements the registrar.Verifier interface
+// returns true if the BMC is identified to be an asrockrack
+func (a *ASRockRack) Compatible() bool {
+
+	resp, statusCode, err := a.queryHTTPS("/", "GET", nil, nil)
+	if err != nil {
+		return false
+	}
+
+	if statusCode != 200 {
+		return false
+	}
+
+	if bytes.Contains(resp, []byte(`ASRockRack`)) {
```

Comment:
you could just return this `bytes.Contains`

Reply:
thanks for pointing these out, fixed in 8c2ffa991